docs/flyte_agents/testing_agents_locally.md

```diff
@@ -0,0 +1,39 @@
+---
+jupytext:
+  formats: md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+---
+
+(testing_agents_locally)=
+# Testing agents locally
+
+Agents can be tested locally without configuring or running the backend server, which makes agent development easier.
+
+The task inherited from `AsyncAgentExecutorMixin` can be executed locally, allowing flytekit to mimic FlytePropeller's behavior to call the agent.
```

Comment:
@pingsutw would you be able to comment with a code snippet I could add here to demonstrate this?

Reply:
https://github.com/flyteorg/flyte/pull/4906